lib/rules/no-warning-comments.js

```diff
@@ -13,44 +13,34 @@ module.exports = function (context) {
 
     var configuration = context.options[0] || {},
         warningTerms = configuration.terms || ["todo", "fixme", "xxx"],
-        location = configuration.location || "start";
+        location = configuration.location || "start",
+        warningRegExps;
 
     /**
-     * Prepares a specified comment for being checked.
-     * @param {String} comment The comment to prepare.
-     * @returns {String} The specified comment prepared for being checked.
+     * Convert a warning term into a RegExp which will match a comment containing that whole word in the specified
+     * location ("start" or "anywhere"). If the term starts or ends with non word characters, then the match will not
+     * require word boundaries on that side.
+     *
+     * @param {String} term A term to convert to a RegExp
+     * @returns {RegExp} The term converted to a RegExp
      */
-    function prepareCommentForChecking(comment) {
-        var commentToCheck;
-
-        commentToCheck = comment.toLowerCase();
-        commentToCheck = commentToCheck.trim();
-
-        return commentToCheck;
-    }
-
-    /**
-     * Checks if the specified comment starts with a specified term.
-     * @param {String} commentToCheck The comment to check.
-     * @param {String} lowerCaseTerm The term to search for.
-     * @returns {Boolean} True if the comment started with the specified term, else false.
-     */
-    function commentStartsWithTerm(commentToCheck, lowerCaseTerm) {
-        return commentToCheck.indexOf(lowerCaseTerm) === 0;
+    function convertToRegExp(term) {
+        var escaped = term.replace(/[-\/\\$\^*+?.()|\[\]{}]/g, "\\$&"),
+            suffix = /\w$/.test(term) ? "\\b" : "",
+            prefix;
+
+        if (location === "start") {
+            prefix = "^\\s*";
+        } else if (/^\w/.test(term)) {
+            prefix = "\\b";
+        } else {
+            prefix = "";
```

Comment:
This is handling the case of a term like `"important!". `\b` only matches the boundary between a word character and a non-word character, so:

```js
/\bimportant!\b/.test("foo important! bar") == false
```